hnn_core/tests/test_drives.py

```diff
@@ -7,19 +7,21 @@
 import numpy as np
 
 import hnn_core
-from hnn_core import Params, Network, read_params
+from hnn_core import Network, read_params
 from hnn_core.drives import (_drive_cell_event_times, _get_prng,
                              _create_extpois, _create_bursty_input)
 from hnn_core.params import _extract_drive_specs_from_hnn_params
 from hnn_core.network import pick_connection
 from hnn_core.network_models import jones_2009_model
 from hnn_core import simulate_dipole
+hnn_core_root = op.dirname(hnn_core.__file__)
 
 
 def test_external_drive_times():
     """Test the different external drives."""
 
-    params = Params()
+    params_fname = op.join(hnn_core_root, 'param', 'gamma_L5weak_L2weak.json')
```

Comment:
The test below assumes the existence of a poisson drive, which only gets added if during `legacy_mode=True` for `default.json`

Reply:
You might be able to get away with removing this `read_params()` step along with `_extract_drive_specs_from_hnn_params` by defining the poisson drive dynamics on L38 explicitly. I think this would be a cleaner unit test that circumvents the black magic contained in `_extract_drive_specs_from_hnn_params`.